src/storage/redact.rs

```diff
@@ -75,6 +128,42 @@ impl From<RedactStorer> for IndexedTypeStorer {
     }
 }
 
+impl RedactStorer {
+    fn get_http_client() -> Result<reqwest::Client, RedactStorerError> {
+        match *ClientTlsConfig::current() {
+            Some(ref ctc) => {
+                let mut pkcs12_vec: Vec<u8> = vec![];
+                File::open(&ctc.pkcs12_path)
+                    .map_err(|source| RedactStorerError::Pkcs12FileNotReadable { source })?
+                    .read_to_end(&mut pkcs12_vec)
+                    .map_err(|source| RedactStorerError::Pkcs12FileNotReadable { source })?;
+                let pkcs12 = reqwest::Identity::from_pem(&pkcs12_vec)
+                    .map_err(|source| RedactStorerError::HttpClientNotBuildable { source })?;
+                let mut ca_cert_vec: Vec<u8> = vec![];
+                File::open(&ctc.server_ca_path)
+                    .map_err(|source| RedactStorerError::ServerCaCertFileNotReadable { source })?
+                    .read_to_end(&mut ca_cert_vec)
+                    .map_err(|source| RedactStorerError::ServerCaCertFileNotReadable { source })?;
+                let ca_cert = reqwest::Certificate::from_pem(&ca_cert_vec)
+                    .map_err(|source| RedactStorerError::HttpClientNotBuildable { source })?;
+                Ok::<_, RedactStorerError>(
+                    reqwest::Client::builder()
+                        .identity(pkcs12)
+                        .add_root_certificate(ca_cert)
+                        .tls_built_in_root_certs(false)
+                        .use_rustls_tls()
+                        .build()
+                        .map_err(|source| RedactStorerError::HttpClientNotBuildable { source })?,
+                )
+            }
+            None => Ok(reqwest::Client::builder()
+                .use_rustls_tls()
+                .build()
+                .map_err(|source| RedactStorerError::HttpClientNotBuildable { source })?),
```

Comment:
what's the expected behavior if there is no config? Will this still work or should it just throw an error?

Reply:
It will work but requests will be anonymous, i.e. will not have client TLS. So requests would fail if they need to be authenticated.